src/maps/eastern_us_and_canada/production.ts

```diff
@@ -0,0 +1,139 @@
+import { SelectAction, SelectData } from "../../engine/select_action/select";
+import { inject, injectState } from "../../engine/framework/execution_context";
+import { Key } from "../../engine/framework/key";
+import { Action } from "../../engine/state/action";
+import z from "zod";
+import { CoordinatesZod } from "../../utils/coordinates";
+import { Good, goodToString, GoodZod } from "../../engine/state/good";
+import { BAG, injectCurrentPlayer } from "../../engine/game/state";
+import { Random } from "../../engine/game/random";
+import { ActionProcessor } from "../../engine/game/action";
+import { GridHelper } from "../../engine/map/grid_helper";
+import { assert } from "../../utils/validate";
+import { City } from "../../engine/map/city";
+import { SelectActionPhase } from "../../engine/select_action/phase";
+import { PHASE } from "../../engine/game/phase";
+import { Phase } from "../../engine/state/phase";
+import { Log } from "../../engine/game/log";
+import { GoodsGrowthPhase } from "../../engine/goods_growth/phase";
+import { PlayerColor } from "../../engine/state/player";
+
+const ProductionState = z.object({
+  goods: GoodZod.array(),
+  cities: CoordinatesZod.array(),
+});
+type ProductionState = z.infer<typeof ProductionState>;
+
+export const PRODUCTION_STATE = new Key("eucProductionState", {
+  parse: ProductionState.parse,
+});
+
+export class EasternUsAndCanadaSelectActionPhase extends SelectActionPhase {
+  private readonly productionState = injectState(PRODUCTION_STATE);
+
+  configureActions(): void {
+    super.configureActions();
+    this.installAction(ProductionAction);
+  }
+  onStart() {
+    super.onStart();
+    this.productionState.initState({ goods: [], cities: [] });
+  }
+  onEnd() {
+    this.productionState.delete();
+    super.onEnd();
+  }
+}
+
+export class EasternUsAndCanadaSelectAction extends SelectAction {
```

Comment:
Do you need to prevent this action if they are in the middle of a production action?

Reply:
Good catch! Without that I think a player could have switched their action after the cube pull.